storage_v2/v1adapter/ptrace_model.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1adapter
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/iter"
+	otel2model "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// PTracesSeq2ToModel consumes an otel trace iterator and returns a jaeger model trace.
+//
+// When necessary, it groups chunks of traces into one single trace
+func PTracesSeq2ToModel(seqTrace iter.Seq2[[]ptrace.Traces, error]) ([]*model.Trace, error) {
```

Comment:
@yurishkuro can this file go in package `jptrace` (potentially in `translator.go`)? I could see this being reused in the future.

Reply:
It feels to me that it makes more sense in v1adapter than in `jptrace` - the latter is only meant to deal with `ptrace` domain objects.

Reply:
@yurishkuro is the translator in the wrong place then? This function looks like its a an extension of the translation functions we defined in `jptrace/translator.go`.

Reply:
Yes, I think translator should also be in v1adapter. Basically the way I am thinking about it is once all internal components are upgraded to work off ptrace model, we should be able to nuke just a couple of folders to remove all of v1 model code.